Resources/ProxyProvider/DIRACCAProxyProvider.py

```diff
@@ -33,227 +31,244 @@ def __init__(self, parameters=None):
     self.match = []
     self.supplied = ['CN']
     self.optional = ['C', 'O', 'OU', 'emailAddress']
+    self.dnList = ['C', 'O', 'OU', 'CN', 'emailAddress']
     # Add not supported distributes names
-    self.fs2nid = X509.X509_Name.nid.copy()
-    self.fs2nid['DC'] = -1
-    self.fs2nid['domainComponent'] = -1
-    self.fs2nid['organizationalUnitName'] = 18
-    self.fs2nid['countryName'] = 14
-    self.n2field = {}  # nid: most short or specidied in CS distributes name
-    self.n2fields = {}  # nid: list of distributes names
+    self.fields2nid = X509.X509_Name.nid.copy()
+    self.fields2nid['DC'] = -1
+    self.fields2nid['domainComponent'] = -1
```

Comment:
Could you please document all that, and use constant name instead of number that seems random ? (-1 , 14, 18, etc)

Reply:
In fact, these are not random numbers, they are nids taken from an M2Crypto.X509.X509_Name object. My rough modification here is to add the missing DC (domainComponent) field that is present in my certificate (which was used for testing) and IIUC it is a rare case. "-1" is here to try to avoid possible conflicts. Other changes are just an extension of the possible names for nids 18(OU) and 14(C).

I'll try to add more doc's about it.